frontend/src/js/util/LruCache.ts

```diff
@@ -85,6 +85,16 @@ export class LruCache<K extends string | number, V> {
   };
 
   replace = (k: K, v: V): void => {
-    this.addToCache(k, v);
+    this.entries[k] = v;
+    this.bumpRecency(k);
+
+    if (this.recencyQueue.length > this.maxEntries) {
```

Comment:
theoretically we shouldn't need this check because replace doesn't increase the length of the queue

Reply:
Oh yes, so we shouldn't need to shift the queue at all, right?